### PR TITLE
[FLINK-1739]Fix the bug of JobManager and TaskManager Port Check.

### PR DESCRIPTION
**Pull request by matadorhong**
The IPC port of JobManager and IPC port of TaskManager must be great than 0 and less than 65535.
It must have check the port whether is in right range area before run the JobManager and TaskManager.

### CODE REVIEW

**Comment by StephanEwen** [comment]
Looks good.

+1 to merge

File: flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala [hunk 1]
```diff
@@ -1274,9 +1274,10 @@ object TaskManager {
         "' is missing (hostname/address of JobManager to connect to).")
     }
 
-    if (port <= 0) {
+    if (port <= 0 || port >= 65535) {
```
Comment (hsaputra) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala]: Should this be 65536 instead of 65535?

Comment (StephanEwen) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala]: Agree, 65535 should be in range...

Comment (matadorhong) [inline flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala]: Yes, It should be in [0, 65535]

**Comment by hsaputra** [comment]
+1 LGTM

**Comment by matadorhong** [comment]
Have modified the port range of TaskManager  in [0, 65535]

**Comment by StephanEwen** [comment]
Good, will merge this in the next batch

**Comment by StephanEwen** [comment]
Merged. Thank you for the patch!

**Pull request closed without merging**